prusti-interface/src/specs/mod.rs

```diff
@@ -299,56 +164,15 @@ impl<'tcx> intravisit::Visitor<'tcx> for SpecCollector<'tcx> {
             self.procedure_specs.insert(local_id, procedure_spec_ref);
         }
 
-        // Collect a typed expression
-        if let Some(expr_id) = read_prusti_attr("expr_id", attrs) {
-            self.typed_expressions.insert(expr_id, local_id);
-        }
-
-        // Collect a specification id and its assertion
+        // Collect spec functions
         if let Some(raw_spec_id) = read_prusti_attr("spec_id", attrs) {
-            let spec_id: SpecificationId = raw_spec_id.try_into()
-                .expect("failed conversion to SpecificationId");
-            let specification = deserialize_spec_from_attrs(attrs);
-
-            // Detect the kind of specification
-            // FIXME: (minor) there is some redundancy here: the type of the
-            // specification doesn't need to be checked. A procedure will refer
-            // to its precondition with a #[pre_spec_id_ref=<id>] attribute,
-            // where <id> is the unique identifier of the specification. Same
-            // for postconditions and invariants.
-            let spec_type = if has_prusti_attr(attrs, "loop_body_invariant_spec") {
-                SpecType::Invariant
-            } else {
-                let fn_name = match fn_kind {
-                    intravisit::FnKind::ItemFn(ref ident, ..) |
-                    intravisit::FnKind::Method(ref ident, ..) => ident.name.to_ident_string(),
-                    intravisit::FnKind::Closure => unreachable!(
-                        "a closure is annotated with prusti::spec_id but not with \
-                        prusti::loop_body_invariant_spec"
-                    ),
-                };
-                if fn_name.starts_with("prusti_pre_item_")
-                    || fn_name.starts_with("prusti_pre_closure_") {
-                    SpecType::Precondition
-                } else if fn_name.starts_with("prusti_post_item_")
-                    || fn_name.starts_with("prusti_post_closure_") {
-                    SpecType::Postcondition
-                } else if fn_name.starts_with("prusti_pred_item_") {
-                    SpecType::Predicate
-                } else {
-                    unreachable!()
-                }
-            };
-
-            let spec_item = SpecItem {spec_id, spec_type, specification};
-            self.spec_items.push(spec_item);
+            let spec_id: SpecificationId = parse_spec_id(raw_spec_id, def_id);
+            self.spec_functions.insert(spec_id, local_id);
 
-            // Collect loop invariant
-            if spec_type == SpecType::Invariant {
-                self.loop_specs
-                    .entry(local_id)
-                    .or_insert(vec![])
-                    .push(spec_id);
+            // Collect loop specifications            
+            if has_prusti_attr(attrs, "loop_body_invariant_spec") {
+                // TODO: surely this isn't all we need to do for loops?
```

Comment:
Is this comment still relevant? This might indeed be all that needs to happen here. Loop specs are then inserted into `def_spec` in the original `determine_loop_specs` function. These are later picked up again in the `ProcedureEncoder`.

Reply:
Addressed by 1bbbdfb4b539f253b77aef890ba056f8df9197e9